src/kademlia/routing_table.cpp

```diff
@@ -171,9 +171,13 @@ void routing_table::status(std::vector<dht_routing_bucket>& s) const
 // TODO: 2 use the non deprecated function instead of this one
 void routing_table::status(session_status& s) const
 {
+	int dht_nodes;
+	int dht_node_cache;
 	int ignore;
-	boost::tie(s.dht_nodes, s.dht_node_cache, ignore) = size();
-	s.dht_global_nodes = num_global_nodes();
+	boost::tie(dht_nodes, dht_node_cache, ignore) = size();
+	s.dht_nodes += dht_nodes;
+	s.dht_node_cache += dht_node_cache;
+	s.dht_global_nodes += num_global_nodes();
```

Comment:
dht_global_nodes actually doesn't make sense to have additive. if anything, it should be the average.. Well, actually, when it's across IPv4 and IPv6, adding would make sense. in the future though, where we may have one DHT node per external interface (which may be multiple of the same address family), then it becomes a bit trickier...